torch/distributed/_shard/partial_tensor.py

```diff
@@ -1,27 +1,24 @@
 import functools
-from typing import Callable, Dict
+from typing import Callable, Dict, TYPE_CHECKING
 
 import torch
 import torch.distributed as dist
 import torch.distributed._shard.sharding_spec as shard_spec
 from torch.distributed import distributed_c10d
-from torch.distributed._shard.sharded_tensor.api import ShardedTensor
 from torch.distributed.nn.functional import (
     reduce_scatter,
 )
+from torch.distributed._shard.common_op_utils import _register_default_op
+from torch.distributed._shard.op_registry_utils import _decorator_func
+from torch.utils._pytree import tree_map
+
+if TYPE_CHECKING:
```

Comment:
it looks like we already do the conditional imports in the `reshard` function, maybe we don't need this import?

Reply:
The "ShardedTensor" return annotation for `reshard` needs this logic.